### PR TITLE
Add staging job for new Windows Tests

### PR DESCRIPTION
**Pull request by adelina-t**
With more tests being added to the main windows release job, we should have a staging job where we can experiment with adding new tests / new configurations without affecting the results of the main job.

### CODE REVIEW

**Comment by PatrickLang** [comment]
Can you also add a few cases that pass consistently so we can make sure the whole pass isn't broken? That would be helpful so we can test changes to the aks-engine apimodel or network plugins in staging.

I think a set like this would be good to test basic scheduling, networking, and a volume

```
[sig-network] DNS should provide DNS for services [Conformance]
[sig-storage] Secrets should be consumable from pods in volume [NodeConformance] [Conformance]
[k8s.io] Kubelet when scheduling a busybox command in a pod should print the output to logs [NodeConformance] [Conformance]
 [k8s.io] Kubelet when scheduling a busybox command that always fails in a pod should be possible to delete [NodeConformance] [Conformance]
```

File: config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml [hunk 1]
```diff
@@ -72,3 +72,43 @@ periodics:
...
+      - --
+      - "--test=true"
+      - "--up=true"
+      - "--down=true"
+      - "--deployment=acsengine"
+      - "--provider=skeleton"
+      - "--build=bazel"
+      - "--acsengine-location=westus2"
+      - "--acsengine-admin-username=azureuser"
+      - "--acsengine-admin-password=AdminPassw0rd"
```
Comment (michmike) [inline config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml]: why is the password in plain text here when we have the environment variable $azure_credentials available? this is duplicated once more in this file.

**Comment by michmike** [comment]
i am approving this and adding a hold as well.
you can remove the hold once you see the question i asked above.
/hold

File: config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml (hunk 1 above)
Comment (PatrickLang) [reply config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml]: that's actually different. $azure_credentials has what's needed to create the VMs.

This username/password can be used to log into the agent VMs. Those are behind a firewall and not accessible. The next step I'm working on is to switch this off password auth and create single-use SSH keypairs for the agent nodes.

Comment (adelina-t) [reply config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml]: @PatrickLang Yes, pretty much like Patrick said: $azure_credentials is used to store info about Azure subscription etc. There is no point in having test-infra folks create and maintain a secret for this. The clusters are generated for a job and destroyed afterwards, the Windows nodes are not publicly accessible so there is not much one can do with this password.

**Comment by adelina-t** [comment]
Updated image version and rebased. @PatrickLang @michmike can you take another look?

@BenTheElder  This change involves updating test-grid. Can you take a look?

**Comment by spiffxp** [comment]
/approve
Looks good from a naming perspective.  The way credentials are seemingly passed in plaintext raises my eyebrows as well.  It might be nice to your future maintainers to comment in the file vs. on the PR why those aren't as bad as they appear.

I'll leave the /lgtm to someone else

**Comment by adelina-t** [comment]
Updated name of template used for staging job.

File: config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml [hunk 2]
```diff
@@ -34,15 +34,15 @@ presets:
 
 periodics:
 - interval: 6h
-  name: ci-kubernetes-e2e-win-1-13
+  name: ci-kuberentes-e2e-aks-engine-azure-master-windows
```
Comment (PatrickLang) [inline config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml]: `ci-kuberentes-e2e-aks-engine-azure-master-windows` has a misspelling throughout the file

**Comment by PatrickLang** [comment]
The job names have `kuberentes` throughout this file. Can you correct that spelling?

**Pull request merged**